c++/CMakeLists.txt

```diff
@@ -64,6 +64,58 @@ elseif (WITH_OPENSSL)
   find_package(OpenSSL REQUIRED COMPONENTS Crypto SSL)
 endif()
 
+set(WITH_FIBERS "AUTO" CACHE STRING
+  "Whether or not to build libkj-async with fibers")
+# define list of values GUI will offer for the variable
+set_property(CACHE WITH_FIBERS PROPERTY STRINGS AUTO ON OFF)
+
+# CapnProtoConfig.cmake.in needs this variable.
+set(_WITH_LIBUCONTEXT OFF)
+
+if (WITH_FIBERS OR WITH_FIBERS STREQUAL "AUTO")
+  set(_capnp_fibers_found OFF)
+  if (WIN32 OR CYGWIN)
+    set(_capnp_fibers_found ON)
+  else()
+    # Fibers need makecontext, setcontext, getcontext, swapcontext that may be in libc,
+    # or in libucontext (e.g. for musl)
+    include(CheckCXXSourceCompiles)
+
+    check_cxx_source_compiles("
+#include <ucontext.h>
+int main() {
+  ucontext_t cp;
+  getcontext(&cp);
+  setcontext(&cp);
```

Comment:
@Sarcasm: I personally like the previous syntax more than this one, but I follow you if you say this is better :blush:. This said, shouldn't we test for swapcontext and makecontext, too?

Reply:
The reason I find this better, is that, one of the reasons CMake configure time can be slow, is because of slow features detection, so I thought a one-shot feature detection would be better than 4, but I could totally be wrong since symbol detection may not require to spawn a compiler (although I checked and it seems to invoke a compiler).

Other reasons is that the CMake configure output shows 2 lines instead of 8, and there is 1 resulting CACHE variable to check instead of 4.

You are right `swapcontext` and/or `makecontext` should probably be checked too, I was lazy (they seem more complex invoke with a function pointer).

I added a new commit that tests them:

- https://github.com/Sarcasm/capnproto/commit/4c54f395a30f8350c89eaa2a991beebb5b77fa17